src/App.tsx

```diff
@@ -15,47 +16,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+function getReorderedGoods(
+  goods: string[],
+  reverse: boolean,
+  sort: SortType,
+): string[] {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sort) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (reverse) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setSortType] = useState(SortType.NONE);
+
+  const handleReverse = () => {
+    setReversed(!isReversed);
+  };
+
+  const handleAlphabetSort = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const handleLengthSort = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const handleReset = () => {
+    setSortType(SortType.NONE);
+    setReversed(false);
+  };
+
+  const reorderedGoods = getReorderedGoods(
+    goodsFromServer,
+    isReversed,
+    sortType,
+  );
+
+  const isVisibleBtn = (isReversed || sortType !== SortType.NONE);
```

Comment:
do not shorten the expression
```suggestion
  const isVisibleButton = (isReversed || sortType !== SortType.NONE);
```
or just `isVisible`